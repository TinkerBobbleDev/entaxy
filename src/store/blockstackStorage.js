/* eslint-disable no-console */
import { getFile, putFile, isUserSignedIn } from 'blockstack'

const filename = 'entaxy.json'

export const loadState = () => {
  if (isUserSignedIn()) {
    return getFile(filename)
      .then(data => JSON.parse(data))
      .catch((error) => {
        console.log(`Error loading file ${filename}:`, error)
        return {}
      })
  }
  return Promise.resolve({})
}

export const saveState = (state) => {
  if (isUserSignedIn()) {
    return putFile(filename, JSON.stringify(state))
  }
  return undefined
}
