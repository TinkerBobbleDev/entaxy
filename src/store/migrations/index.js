import migration0 from './0'
import migration1 from './1'
import migration2 from './2'
import migration3 from './3'

export default {
  0: migration0,
  1: migration1,
  2: migration2,
  3: migration3
}
