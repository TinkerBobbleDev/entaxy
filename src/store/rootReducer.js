import { combineReducers } from 'redux'
import userReducer from './user/reducer'
import settingsReducer from './settings/reducer'
import transactionsReducer from './transactions/reducer'
import marketValuesReducer from './marketValues/reducer'

export default combineReducers({
  user: userReducer,
  settings: settingsReducer,
  transactions: transactionsReducer,
  marketValues: marketValuesReducer
})
