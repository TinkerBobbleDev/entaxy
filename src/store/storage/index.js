/* eslint-disable import/no-cycle */
import { snakeCase } from 'lodash'
import * as blockstackStorage from './blockstackStorage'
import * as localStorage from './localStorage'
import { showOverlay, hideOverlay } from '../settings/actions'
import { initialState as settingsInitialState } from '../settings/reducer'
import { initialState as accountsInitialState } from '../accounts/reducer'
import { initialState as transactionsInitialState } from '../transactions/reducer'
import { initialState as exchangeRatesInitialState } from '../exchangeRates/reducer'
import { initialState as budgetInitialState } from '../budget/reducer'

export const loadItem = (itemName, newState) => (dispatch, getState) => {
  let payload = newState
  if (itemName === 'settings') {
    const { settings } = getState()
    if (settings !== null && settings !== undefined) {
      payload = { ...newState, overlayMessage: settings.overlayMessage }
    }
  }
  dispatch({
    type: `LOAD_${snakeCase(itemName).toUpperCase()}`,
    payload
  })
}

export const loadAllItems = (newState) => (dispatch) => {
  [
    'settings',
    'accounts',
    'transactions',
    'exchangeRates',
    'budget'
  ].map((itemName) => dispatch(loadItem(itemName, newState[itemName] || {})))
}


export const loadState = () => async (dispatch, getState) => {
  if (getState().user === undefined) return {}
  let newState
  switch (getState().user.isAuthenticatedWith) {
    case 'guest':
      dispatch(showOverlay('Loading data from localStorage...'))
      newState = await localStorage.loadState()
      break
    case 'blockstack':
      dispatch(showOverlay('Loading data from Blockstack...'))
      newState = await blockstackStorage.loadState()
      break
    // no default
  }
  if (newState === undefined) newState = { settings: settingsInitialState }
  dispatch(loadAllItems({
    ...newState,
    settings: {
      ...newState.settings,
      snackbarMessage: null
    }
  }))
  return dispatch(hideOverlay())
}

export const resetState = () => (dispatch, getState) => {
  const { settings } = getState()

  // Preserve the existing currency and locale
  dispatch(loadAllItems({
    settings: {
      ...settingsInitialState,
      currency: settings.currency,
      locale: settings.locale
    },
    accounts: accountsInitialState,
    transactions: transactionsInitialState,
    exchangeRates: exchangeRatesInitialState,
    budget: budgetInitialState
  }))
}

export const saveState = () => (_, getState) => {
  let { user } = getState()
  if (user === undefined) user = {}
  switch (user.isAuthenticatedWith) {
    case 'guest':
      return localStorage.saveState(getState())
    case 'blockstack':
      return blockstackStorage.saveState(getState())
    default:
      return undefined
  }
}
