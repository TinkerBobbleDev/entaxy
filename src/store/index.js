import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunkMiddleware from 'redux-thunk'
import rootReducer from './rootReducer'

const middlewares = [thunkMiddleware]

if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger')
  middlewares.push(logger)
}

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middlewares))
)

if (module.hot) {
  module.hot.accept(rootReducer, () => {
    store.replaceReducer(rootReducer)
  })
}

export default store
