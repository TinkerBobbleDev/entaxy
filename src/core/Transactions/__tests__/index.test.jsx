import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from '../../../store'
import { TransactionsComponent } from '../'

describe('Transactions', () => {
  const mochDeleteTransactions = jest.fn()
  const mochHandleSort = jest.fn()
  const props = {
    classes: { }
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot with no transactions', () => {
    const component = renderer.create((
      <Provider store={store}>
        <BrowserRouter>
          <TransactionsComponent
            deleteTransactions={mochDeleteTransactions}
            handleSort={mochHandleSort}
            transactions={[]}
            sortBy="createAt"
            sortDirection="DESC"
            classes={{ ...props }}
          />
        </BrowserRouter>
      </Provider>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot with one transaction', () => {
    const component = renderer.create((
      <Provider store={store}>
        <BrowserRouter>
          <TransactionsComponent
            deleteTransactions={mochDeleteTransactions}
            handleSort={mochHandleSort}
            transactions={[{
              id: 3,
              institution: 'Questrade',
              account: 'RRSP',
              type: 'buy',
              ticker: 'VCE.TO',
              shares: '3',
              bookValue: '3',
              createdAt: new Date()
            }]}
            sortBy="createAt"
            sortDirection="DESC"
            classes={{ ...props }}
          />
        </BrowserRouter>
      </Provider>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
