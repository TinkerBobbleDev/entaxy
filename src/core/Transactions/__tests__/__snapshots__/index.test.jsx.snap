// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions matches snapshot with no transactions 1`] = `
<div>
  <header
    className="MuiPaper-root-11 MuiPaper-elevation4-17 MuiAppBar-root-2 MuiAppBar-positionStatic-6 MuiAppBar-colorPrimary-9"
  >
    <div
      className="MuiToolbar-root-38 MuiToolbar-regular-40 MuiToolbar-gutters-39 Header-toolbar-1"
    >
      <div
        className="Logo-logo-42"
      >
        <img
          alt="Entaxy logo"
          className="Logo-image-43"
          src="logo.png"
        />
        <a
          className="MuiTypography-root-45 MuiTypography-h6-62 MuiTypography-colorInherit-74 Logo-title-44"
          href="/"
          onClick={[Function]}
        >
          Entaxy
        </a>
      </div>
      <div>
        <a
          aria-current={null}
          className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-sizeSmall-103 MuiButton-colorInherit-101"
          href="/portfolio"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-81"
          >
            Portfolio
          </span>
          <span
            className="MuiTouchRipple-root-109"
          />
        </a>
        <a
          aria-current={null}
          className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-sizeSmall-103 MuiButton-colorInherit-101"
          href="/transactions"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-81"
          >
            Transactions
          </span>
          <span
            className="MuiTouchRipple-root-109"
          />
        </a>
        <a
          aria-current={null}
          className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-sizeSmall-103 MuiButton-colorInherit-101"
          href="/import-transactions"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-81"
          >
            Import Transactions
          </span>
          <span
            className="MuiTouchRipple-root-109"
          />
        </a>
      </div>
      <button
        className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-colorInherit-101"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-81"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-109"
        />
      </button>
    </div>
  </header>
  <div
    className="WithFormik-TransactionDialog--root-117"
  />
  <div
    className="MuiPaper-root-11 MuiPaper-elevation2-15 MuiPaper-rounded-12"
    style={
      Object {
        "height": "calc(100% - 100px)",
      }
    }
  >
    <div
      className="MuiToolbar-root-38 MuiToolbar-regular-40 MuiToolbar-gutters-39 TableToolbar-root-133"
    >
      <div
        className="TableToolbar-title-138"
      >
        <h6
          className="MuiTypography-root-45 MuiTypography-h6-62"
          id="tableTitle"
        >
          Transactions
        </h6>
      </div>
      <div
        className="TableToolbar-spacer-135"
      />
      <div
        className="TableToolbar-actions-136"
      >
        <div
          className="TableToolbar-buttons-137"
        >
          <button
            aria-describedby={null}
            aria-label="New transaction"
            className="MuiButtonBase-root-106 MuiIconButton-root-146"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            title="New transaction"
            type="button"
          >
            <span
              className="MuiIconButton-label-151"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-152"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-109"
            />
          </button>
        </div>
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "height": 0,
            "overflow": "visible",
            "width": 0,
          }
        }
      >
        <div
          aria-colcount={8}
          aria-rowcount={0}
          className="ReactVirtualized__Table"
          role="grid"
          style={Object {}}
        >
          <div
            className="ReactVirtualized__Table__headerRow undefined"
            role="row"
            style={
              Object {
                "height": 25,
                "overflow": "hidden",
                "paddingRight": 0,
                "width": 0,
              }
            }
          >
            <div
              className="ReactVirtualized__Table__headerColumn"
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 40px",
                  "flex": "0 1 40px",
                  "msFlex": "0 1 40px",
                }
              }
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
              >
                <span
                  className="MuiButtonBase-root-106 MuiIconButton-root-146 MuiSwitchBase-root-167 MuiCheckbox-root-161 MuiCheckbox-colorSecondary-166"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-151"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-152"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <input
                      checked={false}
                      className="MuiSwitchBase-input-170"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-109"
                  />
                </span>
              </span>
            </div>
            <div
              aria-label="Institution"
              aria-sort="none"
              className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 120px",
                  "flex": "0 1 120px",
                  "msFlex": "0 1 120px",
                }
              }
              tabIndex={0}
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="Institution"
              >
                Institution
              </span>
            </div>
            <div
              aria-label="Account"
              aria-sort="none"
              className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 100px",
                  "flex": "0 1 100px",
                  "msFlex": "0 1 100px",
                }
              }
              tabIndex={0}
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="Account"
              >
                Account
              </span>
            </div>
            <div
              aria-label="Date"
              aria-sort="none"
              className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 120px",
                  "flex": "0 1 120px",
                  "msFlex": "0 1 120px",
                }
              }
              tabIndex={0}
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="Date"
              >
                Date
              </span>
            </div>
            <div
              className="ReactVirtualized__Table__headerColumn"
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "1 1 200px",
                  "flex": "1 1 200px",
                  "msFlex": "1 1 200px",
                }
              }
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="Description"
              >
                Description
              </span>
            </div>
            <div
              aria-label="In"
              aria-sort="none"
              className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 100px",
                  "flex": "0 1 100px",
                  "msFlex": "0 1 100px",
                }
              }
              tabIndex={0}
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="In"
              >
                In
              </span>
            </div>
            <div
              aria-label="Out"
              aria-sort="none"
              className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 100px",
                  "flex": "0 1 100px",
                  "msFlex": "0 1 100px",
                }
              }
              tabIndex={0}
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="Out"
              >
                Out
              </span>
            </div>
            <div
              className="ReactVirtualized__Table__headerColumn"
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 40px",
                  "flex": "0 1 40px",
                  "msFlex": "0 1 40px",
                }
              }
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
              />
            </div>
          </div>
          <div
            aria-label="grid"
            aria-readonly={true}
            className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
            onScroll={[Function]}
            role="rowgroup"
            style={
              Object {
                "WebkitOverflowScrolling": "touch",
                "boxSizing": "border-box",
                "direction": "ltr",
                "height": -25,
                "overflowX": "hidden",
                "overflowY": "auto",
                "position": "relative",
                "width": 0,
                "willChange": "transform",
              }
            }
            tabIndex={0}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Transactions matches snapshot with one transaction 1`] = `
<div>
  <header
    className="MuiPaper-root-11 MuiPaper-elevation4-17 MuiAppBar-root-2 MuiAppBar-positionStatic-6 MuiAppBar-colorPrimary-9"
  >
    <div
      className="MuiToolbar-root-38 MuiToolbar-regular-40 MuiToolbar-gutters-39 Header-toolbar-1"
    >
      <div
        className="Logo-logo-42"
      >
        <img
          alt="Entaxy logo"
          className="Logo-image-43"
          src="logo.png"
        />
        <a
          className="MuiTypography-root-45 MuiTypography-h6-62 MuiTypography-colorInherit-74 Logo-title-44"
          href="/"
          onClick={[Function]}
        >
          Entaxy
        </a>
      </div>
      <div>
        <a
          aria-current={null}
          className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-sizeSmall-103 MuiButton-colorInherit-101"
          href="/portfolio"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-81"
          >
            Portfolio
          </span>
          <span
            className="MuiTouchRipple-root-109"
          />
        </a>
        <a
          aria-current={null}
          className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-sizeSmall-103 MuiButton-colorInherit-101"
          href="/transactions"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-81"
          >
            Transactions
          </span>
          <span
            className="MuiTouchRipple-root-109"
          />
        </a>
        <a
          aria-current={null}
          className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-sizeSmall-103 MuiButton-colorInherit-101"
          href="/import-transactions"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-81"
          >
            Import Transactions
          </span>
          <span
            className="MuiTouchRipple-root-109"
          />
        </a>
      </div>
      <button
        className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-colorInherit-101"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-81"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-109"
        />
      </button>
    </div>
  </header>
  <div
    className="WithFormik-TransactionDialog--root-117"
  />
  <div
    className="MuiPaper-root-11 MuiPaper-elevation2-15 MuiPaper-rounded-12"
    style={
      Object {
        "height": "calc(100% - 100px)",
      }
    }
  >
    <div
      className="MuiToolbar-root-38 MuiToolbar-regular-40 MuiToolbar-gutters-39 TableToolbar-root-133"
    >
      <div
        className="TableToolbar-title-138"
      >
        <h6
          className="MuiTypography-root-45 MuiTypography-h6-62"
          id="tableTitle"
        >
          Transactions
        </h6>
      </div>
      <div
        className="TableToolbar-spacer-135"
      />
      <div
        className="TableToolbar-actions-136"
      >
        <div
          className="TableToolbar-buttons-137"
        >
          <button
            aria-describedby={null}
            aria-label="New transaction"
            className="MuiButtonBase-root-106 MuiIconButton-root-146"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            title="New transaction"
            type="button"
          >
            <span
              className="MuiIconButton-label-151"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-152"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-109"
            />
          </button>
        </div>
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "height": 0,
            "overflow": "visible",
            "width": 0,
          }
        }
      >
        <div
          aria-colcount={8}
          aria-rowcount={1}
          className="ReactVirtualized__Table"
          role="grid"
          style={Object {}}
        >
          <div
            className="ReactVirtualized__Table__headerRow undefined"
            role="row"
            style={
              Object {
                "height": 25,
                "overflow": "hidden",
                "paddingRight": 0,
                "width": 0,
              }
            }
          >
            <div
              className="ReactVirtualized__Table__headerColumn"
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 40px",
                  "flex": "0 1 40px",
                  "msFlex": "0 1 40px",
                }
              }
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
              >
                <span
                  className="MuiButtonBase-root-106 MuiIconButton-root-146 MuiSwitchBase-root-167 MuiCheckbox-root-161 MuiCheckbox-colorSecondary-166"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label-151"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-152"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <input
                      checked={false}
                      className="MuiSwitchBase-input-170"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root-109"
                  />
                </span>
              </span>
            </div>
            <div
              aria-label="Institution"
              aria-sort="none"
              className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 120px",
                  "flex": "0 1 120px",
                  "msFlex": "0 1 120px",
                }
              }
              tabIndex={0}
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="Institution"
              >
                Institution
              </span>
            </div>
            <div
              aria-label="Account"
              aria-sort="none"
              className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 100px",
                  "flex": "0 1 100px",
                  "msFlex": "0 1 100px",
                }
              }
              tabIndex={0}
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="Account"
              >
                Account
              </span>
            </div>
            <div
              aria-label="Date"
              aria-sort="none"
              className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 120px",
                  "flex": "0 1 120px",
                  "msFlex": "0 1 120px",
                }
              }
              tabIndex={0}
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="Date"
              >
                Date
              </span>
            </div>
            <div
              className="ReactVirtualized__Table__headerColumn"
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "1 1 200px",
                  "flex": "1 1 200px",
                  "msFlex": "1 1 200px",
                }
              }
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="Description"
              >
                Description
              </span>
            </div>
            <div
              aria-label="In"
              aria-sort="none"
              className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 100px",
                  "flex": "0 1 100px",
                  "msFlex": "0 1 100px",
                }
              }
              tabIndex={0}
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="In"
              >
                In
              </span>
            </div>
            <div
              aria-label="Out"
              aria-sort="none"
              className="ReactVirtualized__Table__headerColumn ReactVirtualized__Table__sortableHeaderColumn"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 100px",
                  "flex": "0 1 100px",
                  "msFlex": "0 1 100px",
                }
              }
              tabIndex={0}
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
                title="Out"
              >
                Out
              </span>
            </div>
            <div
              className="ReactVirtualized__Table__headerColumn"
              role="columnheader"
              style={
                Object {
                  "WebkitFlex": "0 1 40px",
                  "flex": "0 1 40px",
                  "msFlex": "0 1 40px",
                }
              }
            >
              <span
                className="ReactVirtualized__Table__headerTruncatedText"
              />
            </div>
          </div>
          <div
            aria-label="grid"
            aria-readonly={true}
            className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
            onScroll={[Function]}
            role="rowgroup"
            style={
              Object {
                "WebkitOverflowScrolling": "touch",
                "boxSizing": "border-box",
                "direction": "ltr",
                "height": -25,
                "overflowX": "hidden",
                "overflowY": "auto",
                "position": "relative",
                "width": 0,
                "willChange": "transform",
              }
            }
            tabIndex={0}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
